com.unity.netcode.gameobjects/Tests/Runtime/MultiInstanceHelpers.cs

```diff
@@ -15,24 +15,28 @@ namespace Unity.Netcode.RuntimeTests
     public static class MultiInstanceHelpers
     {
         public const int DefaultMinFrames = 1;
-        public const int DefaultMaxFrames = 64;
+        public const float DefaultTimeout = 1f;
         private static List<NetworkManager> s_NetworkManagerInstances = new List<NetworkManager>();
         private static Dictionary<NetworkManager, MultiInstanceHooks> s_Hooks = new Dictionary<NetworkManager, MultiInstanceHooks>();
         private static bool s_IsStarted;
         private static int s_ClientCount;
         private static int s_OriginalTargetFrameRate = -1;
 
-        public delegate bool MessageReceiptCheck(object receivedMessage);
+        public delegate bool MessageHandleCheck(object receivedMessage);
 
-        private class MultiInstanceHooks : INetworkHooks
+        internal class MessageHandleCheckWithResult
         {
-            public bool IsWaiting;
+            public MessageHandleCheck Check;
+            public bool Result;
+        }
 
-            public MessageReceiptCheck ReceiptCheck;
+        private class MultiInstanceHooks : INetworkHooks
```

Comment:
this one fixes some bugs related to WaitForMessageOfType, so we should get at least changes in this file `MultiInstanceHelpers.cs`

Reply:
for other changes, they might be tricky to get back into `develop` — @NoelStephensUnity might help us here.